vcr/cassette.py

```diff
@@ -87,16 +88,21 @@ def _execute_function(self, function, args, kwargs):
         return handler(function, args, kwargs)
 
     def _handle_coroutine(self, function, args, kwargs):
+        """Wraps a bi-directional generator so that we're inside the cassette
+        context for the duration of the generator.
+        """
         with self as cassette:
             coroutine = self.__handle_function(cassette, function, args, kwargs)
-            to_send = None
+            to_yield = next(coroutine)
             while True:
                 try:
-                    to_yield = coroutine.send(to_send)
-                except StopIteration:
-                    break
-                else:
                     to_send = yield to_yield
+                except Exception:
+                    to_yield = coroutine.throw(*sys.exc_info())
+                else:
+                    to_yield = coroutine.send(to_send)
+            # We don't need to catch StopIteration. The caller (Tornado's
```

Comment:
Why is this comment at a different indentation level. Let's put it above `to_yield = `